swarooppatilx/scruter
# Included chatbot codes for deployment

What does this PR do?

This PR implements a chatbot feature for Scruter. The chatbot is designed to assist users with information about Scruter, such as discovering local services, posting ads, and buying or selling goods within the local community. The chatbot aims to improve user experience by offering instant assistance on the platform's features.

Fixes #<466>

Key Changes

Added chatbot frontend code (HTML, CSS, JavaScript) to enable user interaction within the platform.
Created a backend API in Python (Flask) to handle responses based on predefined Scruter-related queries.
Integrated chatbot responses for common queries related to services, posting ads, buying/selling, and contributing.

Type of change

 Bug fix (non-breaking change which fixes an issue)
 Chore (refactoring code, technical debt, workflow improvements)
 New feature (non-breaking change which adds functionality)
 Breaking change (fix or feature that would cause existing functionality to not work as expected)
 This change requires a documentation update

How should this be tested?

Install Requirements: Follow the instructions in the README.md file to install necessary dependencies.
Run the Backend Server: Execute the Flask app and ensure it’s running on the correct port.
Open the Chatbot on the Frontend: Access the chatbot UI through the web app and test different inputs related to Scruter services, ad postings, and buying/selling.
Confirm Correct Responses: Check if the chatbot gives appropriate responses to user inputs.

Tests to Run

 Test A: Verify that the chatbot responds to queries about local services.
 Test B: Check if ad-related queries trigger appropriate responses.
 Test C: Test input for buying and selling to confirm correct guidance is provided.
 Test D: Evaluate chatbot behavior when querying about contributing to Scruter.

Mandatory Tasks

 Self-reviewed the code and confirmed it aligns with project requirements.
 Updated README.md with installation and deployment instructions.
 Added any necessary documentation updates.
